wandb/sdk/service/server_sock.py

```diff
@@ -0,0 +1,257 @@
+import queue
+import socket
+import threading
+import time
+from typing import Any, Callable, Dict, Optional
+from typing import TYPE_CHECKING
+
+from wandb.proto import wandb_server_pb2 as spb
+
+from .streams import _dict_from_pbmap
+from .streams import StreamMux
+from ..lib.sock_client import SockClient
+
+
+if TYPE_CHECKING:
+    from threading import Event
+    from ..interface.interface_relay import InterfaceRelay
+
+
+class ClientDict:
+    _client_dict: Dict[str, SockClient]
+    _lock: threading.Lock
+
+    def __init__(self) -> None:
+        self._client_dict = {}
+        self._lock = threading.Lock()
+
+    def get_client(self, client_id: str) -> Optional[SockClient]:
+        with self._lock:
+            client = self._client_dict.get(client_id)
```

Comment:
do you need a default value here?

Reply:
No. it is a required argument,  you cant get the client without a client_id

Reply:
sorry I meant in the case that the dict doesn't have the `client_id`.

Reply:
it is an optional response so the default is None.